lib/utils/DevServerPlugin.js

```diff
@@ -35,36 +35,44 @@ class DevServerPlugin {
     // TODO show warning about this
     const isSockJSType = options.webSocketServer.type === 'sockjs';
 
-    /** @type {string} */
-    let host = '';
+    let webSocketURL = '';
+
+    if (typeof options.webSocketURL === 'string') {
+      webSocketURL = options.webSocketURL;
```

Comment:
We need normalize `webSocketURL`, i.e. parse and get `host`/`port`/`path`

Reply:
Same for below code

Reply:
fixed